knative-operator/pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -24,6 +25,13 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
+const (
+	// TODO: The namespace should be editable.
+	ingressNamespace        = "knative-serving-ingress"
+	knativeServingName      = "knative-serving"
+	knativeServingNamespace = "knative-serving"
```

Comment:
Hmm, I think we cannot rely on this (as I've been informed by @jcrossley3 that we don't actually validate the name and the namespace is actually in "flux" as well in theory).

I think we should instead do the "right thing" below as follows:
1. Create a "NamespaceFromIngressNamespace" function that transforms "knative-serving-ingress" to "knative-serving" (essentially a `strings.TrimSuffix` operation I think).
2. List the KnativeServings in that namespace and return the first one (we guarantee that only one is allowed now).

Reply:
I see. I think that is the best way. I tried to implement it.